### ISSUE TITLE
Support rayon::iter::IntoParallelIterator in Reader and Builder as a feature

### ISSUE THREAD

**Issue by mmacedoeu** [capnproto/capnproto-rust#75]
As I see capnproto is all about non copy/ in-place casting so If at any point you need to deep copy all effort is lost. https://github.com/nikomatsakis/rayon is a parallel crate with easy parallel processing but to use it with capnproto without support you need to copy everything losing capnproto purpose.

**Comment by dwrensha** [comment]
I agree that such support for parallel processing sounds useful! I don't think I'll have time in the near future to investigate exactly what would be involved; if you look into it and figure out some details, keep us posted!
